contracts/strategy/ConverterStrategyBase.sol

```diff
@@ -0,0 +1,332 @@
+  function _withdrawFromPoolUniversal(uint amount, bool emergency) internal {
+    if (amount == 0) return;
+
+    if (!emergency) _doHardWork(false);
+
+    address _asset = asset;
+    uint assetBalanceBefore = _balance(_asset);
+
+    if (emergency) {
+      _depositorEmergencyExit();
+    } else {
+      uint liquidityAmount = amount * _depositorLiquidity() / _investedAssets;
+      _depositorExit(liquidityAmount);
+    }
+
+    address[] memory tokens = _depositorPoolAssets();
+    uint len = tokens.length;
+
+    for (uint i = 0; i < len; ++i) {
+      address borrowedToken = tokens[i];
+      if (_asset != borrowedToken) {
+        _closePosition(_asset, borrowedToken, _balance(borrowedToken));
+      }
+    }
+
+    uint amountReceived = _balance(_asset) - assetBalanceBefore;
+    _investedAssets -= amountReceived;
+
+  }
+
+  /// @dev Withdraw given amount from the pool.
+  function _withdrawFromPool(uint amount) override internal virtual {
+    _withdrawFromPoolUniversal(amount, false);
+  }
+
+  /// @dev Withdraw all from the pool.
+  function _withdrawAllFromPool() override internal virtual {
+    _withdrawFromPoolUniversal(_investedAssets, false);
+  }
+
+  /// @dev If pool support emergency withdraw need to call it for emergencyExit()
+  function _emergencyExitFromPool() override internal virtual {
+    _withdrawFromPoolUniversal(_investedAssets, true);
+  }
+
+
+  function _recycle(address[] memory tokens, uint[] memory amounts) internal {
+    require(tokens.length == amounts.length, "SB: Arrays mismatch");
+    uint len = tokens.length;
+
+    for (uint i = 0; i < len; ++i) {
+      address token = tokens[i];
+      uint amount = amounts[i];
+
+      if (amount > 0) {
+        uint amountToCompound = amount * compoundRatio / COMPOUND_DENOMINATOR;
+        if (amountToCompound > 0) {
+          tetuLiquidator.liquidate(token, asset, amountToCompound, LIQUIDATION_SLIPPAGE);
+        }
+
+        uint amountToForward = amount - amountToCompound;
+        if (amountToForward > 0) {
+          _sendToForwarder(token, amount);
+        }
+      }
+    }
+  }
+
+  /// @dev Claim all possible rewards.
+  function _claim() override internal virtual {
+    address[] memory tokens;
+    uint[] memory amounts;
+
+    (tokens, amounts) = _depositorClaimRewards();
+    _recycle(tokens, amounts);
+
+    (tokens, amounts) = tetuConverter.claimRewards(address(this));
+    _recycle(tokens, amounts);
+  }
+
+  /// @dev Is strategy ready to hard work
+  function isReadyToHardWork()
+  override external view returns (bool) {
+    return _isReadyToHardWork;
+  }
+
+  /// @dev Do hard work
+  function doHardWork()
+  override external returns (uint earned, uint lost) {
+    return _doHardWork(true);
+  }
+
+  /// @dev Do hard work
+  function _doHardWork(bool doDepositToPool)
+  internal returns (uint earned, uint lost) {
+
+    uint assetBalanceBefore = _balance(asset);
+    _claim();
+    earned = _balance(asset) - assetBalanceBefore;
```

Comment:
it is not true, you claimed not underlying tokens
need to use a path from liquidator to calc the price in underlying value

Reply:
At _recycle() (used at _claim()) we liquidate compoundRatio rewards to underlying asset.
Only amount sent to Forwarder is not calculated as "earned", but I guess it is should not be added to. Please advice.